modules/telemetry/src/general/nodestelemetry.cpp

```diff
@@ -147,35 +143,61 @@ NodesTelemetry::PrecisionProperties::PrecisionProperties(
     addProperty(highAnglePrecision);
 }
 
-void NodesTelemetry::sendData(int nodeIndex) {
-    if (nodeIndex < 0 || _nodes.size() <= nodeIndex) {
-        LWARNING(std::format("Node list does not include index {}", nodeIndex));
+void NodesTelemetry::update(const Camera* camera) {
+    if (!_enabled) {
         return;
     }
 
-    std::string label = "/" + _nodes[nodeIndex].identifier;
-    std::vector<OpenSoundControlDataType> data(NumDataItems);
+    const SceneGraphNode* focusNode =
+        global::navigationHandler->orbitalNavigator().anchorNode();
 
-    // Distance
-    data[DistanceIndex] = _nodes[nodeIndex].data[DistanceIndex];
+    if (!focusNode) {
+        // Scene is likely not yet initialized
+        return;
+    }
 
-    // Horizontal Angle
-    data[HorizontalAngleIndex] = _nodes[nodeIndex].data[HorizontalAngleIndex];
+    // Get the current angle settings
+    TelemetryModule* module = global::moduleEngine->module<TelemetryModule>();
+    if (!module) {
+        LERROR("Could not find the TelemetryModule");
+        return;
+    }
+    TelemetryModule::AngleCalculationMode angleMode = module->angleCalculationMode();
+    bool includeElevation = module->includeElevationAngle();
 
-    // Vertical Angle
-    data[VerticalAngleIndex] = _nodes[nodeIndex].data[VerticalAngleIndex];
+    // Update data for all nodes
+    for (int i = 0; i < _nodes.size(); ++i) {
+        // Increase precision if the node is in focus
+        if (focusNode->identifier() == _nodes[i].identifier) {
```

Comment:
```suggestion
    for (const TelemetryNode& node : _nodes) {
        // Increase precision if the node is in focus
        if (focusNode->identifier() == node.identifier) {
```

Reply:
I need the node index when calling the `updateData` function further down:
`bool dataWasUpdated = updateData(camera, i, angleMode, includeElevation);`
So I want to keep it as is instead of having to introduce a counter to keep track of the index anyway.